risc0/zkvm/src/guest/env.rs

```diff
@@ -532,6 +564,12 @@ impl<F: Fn(&[u8])> WordWrite for FdWriter<F> {
     }
 }
 
+impl<F: Fn(&[u8])> Drop for FdWriter<F> {
+    fn drop(&mut self) {
+        self.flush();
```

Comment:
I agree that if there are multiple FdWriters for the same resource---which is completely possible, as a user just needs to call env::journal twice---this would actually cause a problem.

Let me think about if this is even fixable... I just look like would require an additional layer of redirection for the sys_write...

Reply:
<- if we want to implement that on the sys_write level, we would be doing something similar to 

```rust
static mut ASSUMPTIONS_DIGEST: MaybePruned<Assumptions> = MaybePruned::Pruned(Digest::ZERO);
```

Reply:
Now obsolete in version 3.